grails-bom/build.gradle

```diff
@@ -1,108 +1,124 @@
-apply plugin: 'maven-publish'
+plugins {
+    id 'java-platform'
+    id 'maven-publish'
+}
 
-publishing {
-    publications {
-        maven(MavenPublication) {
-            pom.withXml {
-                def xml = asNode()
-                def plugins = new Properties()
-                plugins.load(new StringReader(new File("$projectDir/plugins.properties").text))
+javaPlatform {
+    allowDependencies()
+}
 
-                def profiles = new Properties()
-                profiles.load(new StringReader(new File("$projectDir/profiles.properties").text))
+def plugins = new Properties()
+plugins.load(new StringReader(new File("$projectDir/plugins.properties").text))
 
-                xml.children().last() + pomInfo
-                xml.children().last() + {
-                    def mkp = delegate
+def profiles = new Properties()
+profiles.load(new StringReader(new File("$projectDir/profiles.properties").text))
 
-                    mkp.properties {
-                        for (dep in plugins) {
-                            String version = dep.value
-                            if (!isBuildSnapshot && version.endsWith("-SNAPSHOT")) {
-                                throw new RuntimeException("Cannot have a snapshot dependency on a plugin [$dep.key] for a release!")
-                            }
-                            mkp."${dep.key}.version"(version)
-                        }
+ext {
+    // When making changes in the dependencyVersions, remember to also update the Grails BOM Documentation:
+    // https://docs.grails.org/snapshot/ref/Dependency%20Versions/Grails%20BOM.html
+    dependencyVersions = """\
+        com.fasterxml.jackson.core:jackson-databind::$jacksonVersion
+        com.github.ben-manes.caffeine:caffeine::$caffeineVersion
+        com.github.javaparser:javaparser-core::$javaParserCoreVersion
+        com.h2database:h2::$h2Version
+        commons-codec:commons-codec::$commonsCodecVersion
+        hsqldb:hsqldb::$hsqldbVesion
+        io.methvin:directory-watcher::$directoryWatcherVersion
+        jakarta.annotation:jakarta.annotation-api::$jakartaAnnotationApiVersion
+        jakarta.inject:jakarta.inject-api::$jakartaInjectVersion
+        jakarta.persistence:jakarta.persistence-api::$jakartaPersistenceVersion
+        jakarta.servlet:jakarta.servlet-api::$servletApiVersion
+        jakarta.xml.bind:jakarta.xml.bind-api::$jakartaXmlBindVersion
```

Comment:
Some of these use `Api` in the version variable name and some not. Should we be consistent?

Reply:
I did not write these variable names. I am just using the existing variables. I have not modified gradle.properties.

Reply:
Yeah, I know. But this might be a good opportunity change these, if we want.

Reply:
On a side note, do we even need `gradle.properties` versions anymore?  Would the bot be smart enough to update the version here in this proprietary format?

Reply:
> Would the bot be smart enough to update the version here in this proprietary format?

That would surprise me.

Reply:
@matrei  I see no point in keeping versions in `gradle.properties` that are only used once in this file. thoughts?

Reply:
I'm not even sure the bots will be able to update them when they are not used for "real" dependency declarations.
But I think it's important we get that automation working.

We could set up a version catalog for the bots to read and create update PRs on and use those versions in `grails-bom`.
Then they are a bit tucked away from the main build logic.